laiaaa0/assignment3/q1.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include "TestingFunctions.cpp"
+
+int directions_x []= {1,-1,0,0,1,-1,-1,1};
+int directions_y []= {0,0,1,-1,1,1,-1,-1};
+
+bool IsInGrid (int i, int j,const std::vector<std::vector<char> > &grid){
+    if (i>=0 && i<grid.size()){
```

Comment:
This entire expression can actually be a single `return`-statement.
Sometimes that might be at the cost of readability, but in this case, I think it will be reasonably readable.
If you want to make it more readable, you can also have a helper function `InRange(value, min_value, max_value)` that can help make your intention clearer.